### PR TITLE
Fix files changes filtering in CI

### PR DESCRIPTION
**Pull request by frouioui**
## Description

This Pull Request fixes the issue described in https://github.com/vitessio/vitess/issues/11745. `go.sum` and `go.mod` are now two distinct filters in the workflows.

## Related Issue(s)

Fixes https://github.com/vitessio/vitess/issues/11745

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -14,7 +14,7 @@ cloud.google.com/go v0.56.0/go.mod h1:jr7tqZxxKOVYizybht9+26Z/gUq7tiRzu+ACVAMbKV
 cloud.google.com/go v0.57.0/go.mod h1:oXiQ6Rzq3RAkkY7N6t3TcE6jE+CIBBbA36lwQ1JyzZs=
 cloud.google.com/go v0.62.0/go.mod h1:jmCYTdRCQuc1PHIIJ/maLInMho30T/Y0M4hTdTShOYc=
 cloud.google.com/go v0.65.0/go.mod h1:O5N8zS7uWy9vkA9vayVHs65eM1ubvY4h553ofrNHObY=
-cloud.google.com/go v0.72.0/go.mod h1:M+5Vjvlc2wnp6tjzE102Dw08nGShTscUx2nZMufOKPI=
+cloud.google.com/go v0.72.0/go.mod h1:M+5Vjvlc2wnp6tjzE102Dw08nGShTscUzerzex2nZMufOKPI=
```
Comment (deepthi) [inline go.sum:17]: you need to revert this change, right?

Comment (frouioui) [reply go.sum:17]: I definitely do :)

Comment (frouioui) [reply go.sum:17]: It was to test changes to `go.sum`

Comment (frouioui) [reply go.sum]: fixed via 81bf5829e0

**Review by deepthi — APPROVED** [review]

**Comment by deepthi** [comment]
We have only 2 code owners for `/test/`. We should add a few more - maybe @deepthi and @mattlord at the directory level + @harshit-gangal for the java files.

**Comment by frouioui** [comment]
> We have only 2 code owners for `/test/`. We should add a few more - maybe @deepthi and @mattlord at the directory level + @harshit-gangal for the java files.

@deepthi Done: https://github.com/vitessio/vitess/pull/11762
